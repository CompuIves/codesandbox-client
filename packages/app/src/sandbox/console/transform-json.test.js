import transformJSON from './transform-json';

describe('transformJSON', () => {
  it('transforms strings properly', () => {
    expect(transformJSON('Hello world')).toMatchSnapshot();
  });
  it('transforms arrays of strings properly', () => {
    expect(
      transformJSON(['Hello world', 'Bonjour le monde'])
    ).toMatchSnapshot();
  });

  it('transforms numbers properly', () => {
    expect(transformJSON(12)).toMatchSnapshot();
  });

  it('transforms arrays of numbers properly', () => {
    expect(transformJSON([12, 46])).toMatchSnapshot();
  });

  it('transforms objects properly', () => {
    expect(
      transformJSON({
        key1: 'Hello',
        key2: 12,
      })
    ).toMatchSnapshot();
  });

  it('transforms arrays of objects properly', () => {
    expect(
      transformJSON([
        {
          key1: 'Hello',
          key2: 'World',
        },
        {
          key1: 'Bonjour',
          key2: 'Le',
          key3: 'Monde',
        },
      ])
    ).toMatchSnapshot();
  });

  it('transforms named functions properly', () => {
    const fn = function hello(a) {
      return a + 1;
    };
    expect(transformJSON(fn)).toMatchSnapshot();
  });

  it('transforms arrow functions properly', () => {
    const fn = a => a + 1;
    expect(transformJSON(fn)).toMatchSnapshot();

    const fn2 = a => {
      const b = 2;
      return a + b;
    };
    expect(transformJSON(fn2)).toMatchSnapshot();
  });

  it('transforms arrays with functions properly', () => {
    const obj = [
      function hello(a) {
        return a + 1;
      },
      b => b + 1,
    ];

    expect(transformJSON(obj)).toMatchSnapshot();
  });

  it('transforms objects with functions properly', () => {
    const obj = {
      fn1: function hello(a) {
        return a + 1;
      },
      fn2: b => b + 1,
    };

    expect(transformJSON(obj)).toMatchSnapshot();
  });

  it('transforms arrays of objects properly', () => {
    const array = [
      {
        a: 12,
        b: 46,
      },
      { c: 'Hello', d: 'World' },
      {
        fn1: function hello(a) {
          return a + 1;
        },
        fn2: b => b + 1,
      },
    ];

    expect(transformJSON(array)).toMatchSnapshot();
  });
});
