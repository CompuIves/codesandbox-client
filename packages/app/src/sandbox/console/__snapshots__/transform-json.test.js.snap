// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformJSON transforms arrays of numbers properly 1`] = `"[12,46]"`;

exports[`transformJSON transforms arrays of objects properly 1`] = `"[{\\"key1\\":\\"Hello\\",\\"key2\\":\\"World\\"},{\\"key1\\":\\"Bonjour\\",\\"key2\\":\\"Le\\",\\"key3\\":\\"Monde\\"}]"`;

exports[`transformJSON transforms arrays of objects properly 2`] = `"[{\\"a\\":12,\\"b\\":46},{\\"c\\":\\"Hello\\",\\"d\\":\\"World\\"},{\\"fn1\\":\\"function hello(a) {\\\\n        return a + 1;\\\\n      }\\",\\"fn2\\":\\"function fn2(b) {return b + 1;}\\"}]"`;

exports[`transformJSON transforms arrays of strings properly 1`] = `"[\\"Hello world\\",\\"Bonjour le monde\\"]"`;

exports[`transformJSON transforms arrays with functions properly 1`] = `"[\\"function hello(a) {\\\\n      return a + 1;\\\\n    }\\",\\"function (b) {return b + 1;}\\"]"`;

exports[`transformJSON transforms arrow functions properly 1`] = `"\\"function fn(a) {return a + 1;}\\""`;

exports[`transformJSON transforms arrow functions properly 2`] = `"\\"function fn2(a) {\\\\n      var b = 2;\\\\n      return a + b;\\\\n    }\\""`;

exports[`transformJSON transforms named functions properly 1`] = `"\\"function hello(a) {\\\\n      return a + 1;\\\\n    }\\""`;

exports[`transformJSON transforms numbers properly 1`] = `"12"`;

exports[`transformJSON transforms objects properly 1`] = `"{\\"key1\\":\\"Hello\\",\\"key2\\":12}"`;

exports[`transformJSON transforms objects with functions properly 1`] = `"{\\"fn1\\":\\"function hello(a) {\\\\n        return a + 1;\\\\n      }\\",\\"fn2\\":\\"function fn2(b) {return b + 1;}\\"}"`;

exports[`transformJSON transforms strings properly 1`] = `"\\"Hello world\\""`;
